test/integration/quic_http_integration_test.cc

```diff
@@ -1735,6 +1737,18 @@ TEST_P(QuicHttpIntegrationTest, UsesPreferredAddress) {
     server_preferred_address.set_ipv4_address("127.0.0.2");
     server_preferred_address.set_ipv6_address("::2");
     preferred_address_config->mutable_typed_config()->PackFrom(server_preferred_address);
+
+    // Configure a test listener filter which is incompatible with any server preferred addresses
+    // but with any matcher, which effectively disables the filter.
+    auto* listener_filter =
+        bootstrap.mutable_static_resources()->mutable_listeners(0)->add_listener_filters();
```

Comment:
so what happened before if we tried to add a QUIC listener filter?  no-op?  config rejection?  I'd expect both proto/API updates with this PR as well as release notes.

Reply:
I think we should at least have release notes unless Ryan disagrees.

Reply:
I added both. If any reviewer have concerns about API stability, I will revert the API doc change.